kernel/src/hil/uart.rs

```diff
@@ -46,6 +46,9 @@ pub enum Error {
 
     /// No error occurred and the command completed successfully
     CommandComplete,
+
+    /// Client requested transfer bigger than buffer
+    ClientRequestLargerThanBuffer,
```

Comment:
This shouldn't be a uart::Error code. The interface semantics should be that it sends/receives the minimum of the passed length and the actual buffer length.